planner/core/rule_column_pruning.go

```diff
@@ -129,6 +129,15 @@ func (ls *LogicalSort) PruneColumns(parentUsedCols []*expression.Column) {
 
 // PruneColumns implements LogicalPlan interface.
 func (p *LogicalUnionAll) PruneColumns(parentUsedCols []*expression.Column) {
+	used := getUsedList(parentUsedCols, p.schema)
+	hasBeenUsed := false
+	for i := range used {
+		hasBeenUsed = hasBeenUsed || used[i]
+	}
+	if !hasBeenUsed {
```

Comment:
what if 1 column is used in parent of Union, so `hasBeenUsed` is true, after the child projection is removed later, Union is a 1-column schema, child Join is a 2-column schema, will the panic still happen?

Reply:
It seems that there will always remain a Projection between Union and Join in this context since ProjectionElimination is called after ColumnPrune.
So the Projection upon Join cannot be eliminated.

Reply:
Hmm, the projection would be kept then, so no panic.